docs/examples/edit_distance.md

```diff
@@ -0,0 +1,121 @@
+# Edit Distance
+
+- Given two strings `str1` and `str2`, what is the minimum number of edits
+  to convert `str1` to `str2`?
+- Each edit costs 1, and is either an add, delete, or replace.
```

Comment:
Let's use the terminology in this [lecture note](https://web.stanford.edu/class/cs124/lec/med.pdf) by Stanford. Also can you briefly explain what each of the operation means (with a little bit of formality)? e.g., it's not obvious to a student what does an add/insertion mean.